traitsui/wx/instance_editor.py

```diff
@@ -527,6 +527,7 @@ def edit_instance(self, event):
             # have its own:
             if ui.history is None:
                 ui.history = self.ui.history
+        return ui
```

Comment:
Following the pattern in the Qt implementation, it might be better to assign `ui` to a private attribute on the object and make sure its `dispose` method is called, this will allow the mouse click to be a separate interaction.